server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -119,6 +122,124 @@ const orderMutations = {
+      },
+      paymentMethod: {
+        type: PaymentMethodReferenceInput,
+        description: 'Object matching the PaymentMethodReferenceInput (uuid)',
+      },
+    },
+    async resolve(_, args, req) {
+      const decodedId = getDecodedId(args.order.id);
+
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to update a subscription');
+      }
+
+      const { paymentMethod } = args;
+
+      const query = {
+        where: {
+          id: decodedId,
+        },
+        include: [{ model: models.Subscription }, { model: models.PaymentMethod, as: 'paymentMethod' }],
+      };
+
+      let order = await models.Order.findOne(query);
+
+      if (!order) {
+        throw new NotFound('Subscription not found');
+      }
+      if (!req.remoteUser.isAdmin(order.FromCollectiveId)) {
+        throw new Unauthorized("You don't have permission to update this subscription");
+      }
+      if (!order.Subscription.isActive) {
+        throw new Error('Subscription must be active to be updated');
+      }
+
+      // payment method
+      if (paymentMethod !== undefined) {
+        // unlike v1 we don't have to check/assign new payment method, that will be taken care of in another mutation
+        const newPaymentMethod = await models.PaymentMethod.findOne({
+          where: { uuid: paymentMethod.uuid },
+        });
+        if (!newPaymentMethod) {
+          throw new Error('Payment method not found with this uuid', paymentMethod.uuid);
+        }
+        if (!req.remoteUser.isAdmin(paymentMethod.CollectiveId)) {
+          throw new Unauthorized("You don't have permission to use this payment method");
+        }
+
+        order = await order.update({ PaymentMethodId: newPaymentMethod.id });
+      }
+
+      return order;
+    },
+  },
+  addPaymentMethod: {
+    type: Order,
+    description: 'Add a new payment method to be used with an Order',
+    args: {
+      newPaymentMethod: {
+        type: NewPaymentMethodInput,
+        description: 'Object matching the NewPaymentMethodInput',
+      },
+    },
+    async resolve(_, args, req) {
+      const collective = await models.Collective.findByPk(req.remoteUser.CollectiveId);
+      if (!collective) {
+        throw Error('This collective does not exist');
+      }
+
+      const { newPaymentMethod } = args;
+
+      const newPaymentMethodData = {
+        ...newPaymentMethod,
+        service: newPaymentMethod.service || 'stripe',
+        CreatedByUserId: req.remoteUser.id,
+        currency: args.currency || collective.currency,
+        saved: true,
+        CollectiveId: req.remoteUser.CollectiveId,
+      };
+
+      let pm = await models.PaymentMethod.create(newPaymentMethodData);
+
+      try {
+        pm = await setupCreditCard(pm, {
+          collective,
+          user: req.remoteUser,
+        });
+      } catch (error) {
+        if (!error.stripeResponse) {
+          throw error;
+        }
+
+        pm.stripeError = {
```

Comment:
Good that you're doing that. It's critical in the previous implementation to confirm cards with something like 3D secure. This is a case, we will need to make sure is working.